src/append/rolling_file/policy/compound/trigger/onstartup.rs

```diff
@@ -0,0 +1,126 @@
+//! The OnStartUp trigger.
+//!
+//! Requires the `onstartup_trigger` feature.
+
+use std::sync::Once;
+
+use crate::append::rolling_file::{policy::compound::trigger::Trigger, LogFile};
+
+#[cfg(feature = "config_parsing")]
+use crate::config::{Deserialize, Deserializers};
+
+/// Configuration for the onstartup trigger.
+#[cfg(feature = "config_parsing")]
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug, Default, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct OnStartUpTriggerConfig {
+    #[serde(default = "default_min_size")]
+    min_size: u64,
+}
+
+#[cfg(feature = "config_parsing")]
+fn default_min_size() -> u64 {
+    1
+}
+
+/// A trigger which rolls the log on startup.
+#[derive(Debug)]
+pub struct OnStartUpTrigger {
+    min_size: u64,
+    initial: Once,
+}
+
+impl OnStartUpTrigger {
+    /// Returns a new trigger which rolls the log on startup.
+    pub fn new(min_size: u64) -> OnStartUpTrigger {
+        OnStartUpTrigger {
+            min_size,
+            initial: Once::new(),
+        }
+    }
+}
+
+impl Trigger for OnStartUpTrigger {
+    fn trigger(&self, file: &LogFile) -> anyhow::Result<bool> {
+        if !self.initial.is_completed() {
```

Comment:
While the chances of this resulting in an observed race are slim, this is technically a race. `initial.call_once()` _could_ have been called between this line and the next, resulting in a double trigger.